riscv/csrs.h

```diff
@@ -637,7 +637,7 @@ class debug_mode_csr_t: public basic_csr_t {
   virtual void verify_permissions(insn_t insn, bool write) const override;
 };
 
-typedef std::shared_ptr<tdata2_csr_t> tdata2_csr_t_p;
+typedef std::shared_ptr<csr_t> tdata2_csr_t_p;
```

Comment:
This typedef is no longer useful and is now misleading because it doesn't point to a `tdata2_csr_t`. Instead simply change any uses of `tdata2_csr_t_p` to use `csr_t_p` instead and delete this typedef.

Reply:
Fixed